pkg/state/state_ofed.go

```diff
@@ -590,3 +593,69 @@ func (s *stateOFED) setEnvFromClusterWideProxy(cr *mellanoxv1alpha1.NicClusterPo
 		)
 	}
 }
+
+// mergeWithDefaultEnvs returns env variables provided in currentEnvs merged with default
+// env variables for MOFED container, taking into account attributes of the k8s cluster.
+func (s *stateOFED) mergeWithDefaultEnvs(
+	currentEnvs []v1.EnvVar, nodeAttrs map[nodeinfo.AttributeType]string) []v1.EnvVar {
+	if envVarsWithGet(currentEnvs).Get("CREATE_IFNAMES_UDEV") != nil {
+		// already exists dont overwrite
+		return currentEnvs
+	}
+
+	// CREATE_IFNAMES_UDEV: should be set to true for ubuntu < 22.04 , RHEL < 9, OCP < 4.13 if not provided
+	osName := nodeAttrs[nodeinfo.AttrTypeOSName]
+	osVer := nodeAttrs[nodeinfo.AttrTypeOSVer]
+	createIfnameUdevEnv := v1.EnvVar{Name: "CREATE_IFNAMES_UDEV", Value: "true"}
+
+	currVer, err := semver.NewVersion(osVer)
+	if err != nil {
+		return currentEnvs
+	}
+
+	ubuntuVer, err := semver.NewVersion("22.04")
```

Comment:
nit: use semver.MustParse
https://github.com/Masterminds/semver/blob/e06051f8fcc4c8b4a4990c337b9862a2448722e5/version.go#L230

Reply:
nice! will rework it.